File: tests/bench/crosslang/c/arith.c
/* Порт tests/bench/programs/ArithBench.eat 1:1 — целочисленная
 * арифметика в горячем цикле. Вывод обязан совпасть байт-в-байт
 * с EATLang-оригиналом при том же REPEAT. */
#include <stdint.h>
#include <stdio.h>

static const uint32_t REPEAT = 1;

static uint32_t work(uint32_t seed) {
    uint32_t acc = seed;
    for (uint32_t k = 0; k < 1000u; k++) {
        for (uint32_t i = 0; i < 1000u; i++) {
            acc = (acc * 31u + i) % 65536u;
        }
    }
    return acc;
}

int main(void) {
    uint32_t acc = 7u;
    for (uint32_t r = 0; r < REPEAT; r++) {
        acc = work(acc) % 65536u;
    }
    printf("checksum %u\n", acc);
    return 0;
}

File: tests/bench/crosslang/rust/arith.rs
// Порт tests/bench/programs/ArithBench.eat 1:1 — целочисленная
// арифметика в горячем цикле. Вывод обязан совпасть байт-в-байт
// с EATLang-оригиналом при том же REPEAT.

const REPEAT: u32 = 1;

fn work(seed: u32) -> u32 {
    let mut acc: u32 = seed;
    for _ in 0..1000u32 {
        for i in 0..1000u32 {
            acc = (acc * 31 + i) % 65536;
        }
    }
    acc
}

fn main() {
    let mut acc: u32 = 7;
    for _ in 0..REPEAT {
        acc = work(acc) % 65536;
    }
    println!("checksum {}", acc);
}
